src/indexer/updater.ts

```diff
@@ -446,13 +446,22 @@ export class RuneUpdater implements RuneBlockIndex {
       const utxoBalance =
         utxoBalancesByOutputLocation.get(`${input.txid}:${input.vout}`) ??
         (await this._storage.getUtxoBalance(input.txid, input.vout));
-      this.spentOutputs.push({ txid: input.txid, vout: input.vout });
       for (const additionalBalance of utxoBalance) {
         const runeId = additionalBalance.runeId;
         const runeLocation = RuneLocation.toString(runeId);
         const balance = unallocated.get(runeLocation) ?? { runeId, amount: 0n };
         unallocated.set(runeLocation, balance);
         balance.amount = u128.checkedAddThrow(u128(balance.amount), u128(additionalBalance.amount));
+        this.spentOutputs.push({
```

Comment:
maybe more accurate to call it `spentBalances` instead of `spentOutputs`

Reply:
Much better